### PR TITLE
create uberjar with scala helpers using sbt

### PR DESCRIPTION
**Pull request by joelgrus**
replaces the `scalac` make step with `sbt`, configured to produce one jar with both the dynet bindings and the scala helpers (and with the examples too)

### CODE REVIEW

File: swig/README.md [hunk 1]
```diff
@@ -15,12 +15,15 @@ the `build` directory (see [DyNet documentation](http://dynet.readthedocs.io/en/
 
 ```
 build$ cmake .. -DEIGEN3_INCLUDE_DIR=../eigen -DINCLUDE_SWIG=ON
-build$ make
+build$ make dynet_swig && make
```
Comment (OyvindTafjord) [inline swig/README.md:18]: It would be good to figure out why this is (sometimes) needed, the intention is that running just `make` should be sufficient.

**Review by OyvindTafjord — APPROVED** [review]
This looks like a good start. Not sure about the build touching things outside the `build` directory, but maybe that's more or less unavoidable with the `sbt` stuff running.

**Comment by joelgrus** [comment]
I just changed it so that instead of copying the jar to `swig/lib` it changes the `unmanagedBase` to point to the build directory. (it still runs sbt in the `swig` directory).

I don't like hardcoding the `../build/swig` path, but passing arguments into sbt is surprisingly complicated, I'll deal with that next.

**Comment by OyvindTafjord** [comment]
Sounds good to me

**Comment by joelgrus** [comment]
ok, latest change makes everything (intermediate class files, final uberjar) happen in the (still hardcoded) build directory, I'll merge and then *now* I'll really work on the command line args

**Pull request merged**